front-end-tools/src/Main.tsx

```diff
@@ -375,6 +386,32 @@ export default function Main(props: ConnectionProps) {
                                                 const file = hTMLInputElement.files[0];
                                                 const arrayBuffer = await file.arrayBuffer();
 
+                                                // Concordium's tooling create versioned modules e.g. `.wasm.v1` now.
+                                                // Unversioned modules cannot be created by Concordium's tooling anymore.
+                                                // We assume that the uploaded module is a versioned module
+                                                // and remove the versioned 8 bytes at the beginning.
+                                                const wasmModule = await WebAssembly.compile(
+                                                    arrayBuffer.slice(8)
+                                                ).catch((e) => {
```

Comment:
Combining `await` and `catch` seems odd. I think you should stick with pure promise `then/catch` or handle error with `try/catch`.